core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -831,11 +830,6 @@ public void run() {
         } else {
           noMoreTransparentRetry.set(true);
 
-          if (retryPolicy == null) {
```

Comment:
It's not immediately clear to me why this block is no longer necessary.  `retryPolicy` was not accessed in this function after this conditional even before this PR - was it previously unnecessary, and this is just cleaning up? Or was the assignment of retryPolicy required before this change and not required now?

Reply:
This block is moved to `makeRetryDecision()`.